numpy/core/src/multiarray/temp_elide.c

```diff
@@ -7,6 +7,7 @@
 #include "numpy/arrayobject.h"
 
 #define NPY_NUMBER_MAX(a, b) ((a) > (b) ? (a) : (b))
+#define ARRAY_SIZE(a) (sizeof(a)/sizeof(a[0]))
```

Comment:
Is there a private internal header this can go into? (can be in a later PR)

Reply:
I thought about that. There didn't seem to be a good place in `numpy/core/include/numpy` or in `numpy/core/src/private`, and `numpy/core/src/multiarray/common.h` didn't seem quite right either. If we want to go that route I think maybe a new `numpy/core/src/private/npy_macros.h` or some such might be the best way to go.